### PR TITLE
bazel: Add python `envoy_entry_point`

### PR DESCRIPTION
**Pull request by phlax**
This extends the `rules_python` `entry_point` to allow data to be
added to the sandbox, and args to pass the data to the tool.

Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
with `py_binary` you can add other rules as `data` and then pass the data to the py binary as `args`

with `entry_point` you are unable to do that, this PR adds this functionality

i opened an upstream issue here https://github.com/bazelbuild/rules_python/issues/600

**Comment by phlax** [comment]
this might be unnecessary, just thinking of another possible way to do it...

**Comment by phlax** [comment]
> ...another possible way to do it...

which didnt work, so i think we want this PR

File: tools/base/envoy_python.bzl [hunk 1]
```diff
@@ -71,3 +72,28 @@ def envoy_py_binary(
...
+        entry_point = base_entry_point,
+        script = None,
+        data = None,
+        args = None):
+    script = script or pkg
+    entry_point_name = "%s_entry_point" % name
+    native.alias(
+        name = entry_point_name,
+        actual = entry_point(
+            pkg = pkg,
```
Comment (htuch) [inline tools/base/envoy_python.bzl:89]: Can you add some macro level documentation on the parameters? Like, why is there both an entry point name and a pkg, and why does the user of this macro care?

Comment (phlax) [reply tools/base/envoy_python.bzl:89]: sure, ill add a docstring

Comment (phlax) [reply tools/base/envoy_python.bzl:105]: done, ill also fixed the proc handling in the dynamic loader script

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**